library/core/src/iter/traits/iterator.rs

```diff
@@ -3316,6 +3316,49 @@ pub trait Iterator {
         Cycle::new(self)
     }
 
+    /// Returns an iterator over `N` elements of the iterator at a time.
+    ///
+    /// The chunks do not overlap. If `N` does not divide the length of the
+    /// iterator, then the last up to `N-1` elements will be omitted and can be
+    /// retrieved from the [`.into_remainder()`][ArrayChunks::into_remainder]
+    /// function of the iterator.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `N` is 0.
```

Comment:
Technically this doesn't panic like this.  I think it probably should -- at least for now, same as <https://doc.rust-lang.org/nightly/std/primitive.slice.html#method.as_chunks> -- so please add an assert for that.

Reply:
Oh, nvm -- I see that it asserts in `new`.